semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -398,19 +398,19 @@ let rec m_name a b =
   | ( a,
       B.Id
         ( idb,
-          {
-            B.id_resolved =
-              {
-                contents =
-                  Some
-                    ( ( B.ImportedEntity dotted
-                      | B.ImportedModule (B.DottedName dotted)
-                      | B.ResolvedName dotted ),
-                      _sid );
-              };
-            _;
-          } ) ) -> (
-      m_name a (B.Id (idb, B.empty_id_info ()))
+          ({
+             B.id_resolved =
+               {
+                 contents =
+                   Some
+                     ( ( B.ImportedEntity dotted
+                       | B.ImportedModule (B.DottedName dotted)
+                       | B.ResolvedName dotted ),
+                       _sid );
+               };
+             _;
+           } as infob) ) ) -> (
+      m_name a (B.Id (idb, { infob with B.id_resolved = ref None }))
```

Comment:
This is subtle; I would have added a comment like
(* bugfix: do not use B.empty_id_info() here because we would lose the type and svalue for the id and this
 * could prevent XXX *)